### DISCUSSION TITLE
Invalid version: '2023.01.09-afd71a6c' (package: searxng)

### DISCUSSION THREAD

**Discussion by ghost** [searxng/searxng#2120] [Q&A] [answered]
I get an error when I try to update searxng using `searxng.sh`.
https://pastebin.com/pchwwkeN
From what I understand, the script 'normalizes' version names with a two-digit month by removing a leading zero. It then tries to use an 'un-normalized' version somewhere later which results in an error. How can I fix this?

**Comment by dalf** [comment]
can you try to replace this line
https://github.com/searxng/searxng/blob/674dae946afe3d14b592b5304b895399722403a8/searx/version.py#L62
by
```python
    git_commit_date_hash = subprocess_run(r"git show -s --date='format:%Y.%-m.%-d' --format='%cd+%h'")
```
?

Notice the additional `-` after `%`. I will look after this issue tomorrow.

**Comment by ghost** [comment]
Okay, I did that. Now when I try to update:
https://pastebin.com/DTH8C8Yv
I have redis 4.4.2, not 4.4.0. What is going on? Should I just reinstall?

**Comment by return42** [comment]
**UPDATE: the issue is a follow up of invalid version identifiers / read https://github.com/searxng/searxng/discussions/2120#discussioncomment-4749860 to solve the issue.**

-----


> Okay, I did that. Now when I try to update:

```
fetch from https://github.com/searxng/searxng and reset to origin/master
...
  |searxng| HEAD is now at 674dae94 Merge pull request #2119 from dalf/weblate_codeberg
...
```
now we are at 674dae94 ..

> I have redis 4.4.2, not 4.4.0.

```
Obtaining file:///usr/local/searxng/searxng-src
...
pip._vendor.pkg_resources.ContextualVersionConflict: 
  (redis 4.4.2 (/usr/local/searxng/searx-pyenv/lib/python3.9/site-packages), 
  Requirement.parse('redis==4.4.0'), {'searxng'})
```

Strange, I do not understand that requirement ``Requirement.parse('redis==4.4.0'), {'searxng'}`` .. when  we look at 674dae94 ... 

https://github.com/searxng/searxng/blob/674dae946afe3d14b592b5304b895399722403a8/requirements.txt#L15

.. redis 4.4.2 is required.

Did you modified files in your installation at `file:///usr/local/searxng/searxng-src`?

> Should I just reinstall?

May be re-install is more trivial compared to dig deeper in this issue .. but before reinstall, I think we have to consolidate our mess around the version identifier first (this was your initial issue) --> https://github.com/searxng/searxng/pull/2122

**Comment by return42** [comment, marked as answer]
We had invalid version identifiers in the past / fixed in https://github.com/searxng/searxng/pull/2122

When you try to update, pip can't uninstall the previous installed version due to the invalid version number that is installed.

A simple workaround is to remove the invalid version identifier manually by removing the searxng.egg-info file from the previous installation (from the last update):

    $ sudo -H rm -rf /usr/local/searxng/searxng-src/searxng.egg-info/
    $ sudo -H ./utils/searxng.sh instance update

**Comment by ghost** [comment]
Thank you very much. I removed the directory and updated to `a8eb9d15` without error.
